idea-plugin/src/main/kotlin/arrow/meta/ide/dsl/editor/fileEditor/EditorSyntax.kt

```diff
@@ -0,0 +1,89 @@
+package arrow.meta.ide.dsl.editor.fileEditor
+
+import arrow.meta.ide.IdeMetaPlugin
+import arrow.meta.ide.phases.application.ApplicationProvider
+import arrow.meta.internal.Noop
+import arrow.meta.phases.ExtensionPhase
+import com.intellij.openapi.editor.Document
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.editor.EditorFactory
+import com.intellij.openapi.editor.event.CaretEvent
+import com.intellij.openapi.editor.event.CaretListener
+import com.intellij.openapi.fileEditor.FileDocumentManager
+import com.intellij.openapi.fileEditor.FileEditor
+import com.intellij.openapi.fileEditor.FileEditorManager
+import com.intellij.openapi.fileEditor.FileEditorManagerEvent
+import com.intellij.openapi.fileEditor.FileEditorManagerListener
+import com.intellij.openapi.fileEditor.FileEditorProvider
+import com.intellij.openapi.util.Pair
+import com.intellij.openapi.vfs.VirtualFile
+
+interface EditorSyntax {
+
+  /**
+   * Registers a [FileEditorManagerListener] in order to add a [CaretListener] for each editor
+   * so foldingRegions can be collapsed once the caret has change its position
+   */
+  fun IdeMetaPlugin.addCaretListener(
+    caretAdded: Editor.(event: CaretEvent) -> Unit = {},
+    caretPositionChanged: Editor.(event: CaretEvent) -> Unit = {},
```

Comment:
Please, add the Document, which the CaretListener is currently observing to all 3 parameters